src/isar/models/mission.py

```diff
@@ -1,37 +1,60 @@
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from datetime import datetime
-from typing import List, Union
-from uuid import UUID
+from typing import Iterator, List, Union
+from uuid import UUID, uuid4
 
 from isar.config.settings import settings
 from isar.models.mission_metadata.mission_metadata import MissionMetadata
-from robot_interface.models.mission import STEPS
-from robot_interface.models.mission.step import DriveToPose, InspectionStep
+from robot_interface.models.mission import STEPS, Step, StepStatus
 
 
 @dataclass
-class Mission:
+class Task:
     steps: List[STEPS]
+    status: StepStatus = field(default=StepStatus.NotStarted, init=False)
+    id: UUID = field(default_factory=uuid4, init=False)
+    iterator: Iterator = None
+
+    def next_step(self) -> Step:
+        return self.iterator.__next__()
+
+    def is_finished(self) -> bool:
+        for step in self.steps:
+            if step.status is StepStatus.Failed or step.status is StepStatus.Unexpected:
+                # One step has failed meaning the task as a whole should be aborted
+                self.status = StepStatus.Failed
```

Comment:
With the current implementation, the task will always fail if a step of the task fails and the whole task cannot be completed. Is this the desired behavior, or do we want the task to continue? In the case of a task with multiple inspection steps, the next inspection step might succeed even if the first inspection step fails.

Reply:
I agree, same discussion as with @tsundvoll above. Conceptually I think it's easier to say that a `Task` should require all steps to succeed, otherwise it will be a nightmare to infere the status of the task based on so many conditions. However, this would make it more natural to have multiple tasks for multiple inspections of the same tag, which again moved the mission away from the input list mission.

Reply:
See conversation above.